pex/pep425tags.py

```diff
@@ -12,345 +15,317 @@
 # please consult `pex.interpreter.ID_PY_TMPL` and adjust stdlib imports there as needed.
 def get_flag(var, fallback, expected=True, warn=True):
-  """Use a fallback method for determining SOABI flags if the needed config
-  var is unset or unavailable."""
-  val = get_config_var(var)
-  if val is None:
-    if warn:
-      logger.debug("Config variable '%s' is unset, Python ABI tag may "
-                   "be incorrect", var)
-    return fallback()
-  return val == expected
+    """Use a fallback method for determining SOABI flags if the needed config
+    var is unset or unavailable."""
+    val = get_config_var(var)
+    if val is None:
+        if warn:
+            log.debug("Config variable '%s' is unset, Python ABI tag may "
+                      "be incorrect", var)
+        return fallback()
+    return val == expected
 
 
 # NB: Used in `pex.interpreter.ID_PY_TMPL` and should only rely on stdlib. If imports change,
 # please consult `pex.interpreter.ID_PY_TMPL` and adjust stdlib imports there as needed.
 def get_abi_tag():
-  """Return the ABI tag based on SOABI (if available) or emulate SOABI
-  (CPython 2, PyPy)."""
-  soabi = get_config_var('SOABI')
-  impl = get_abbr_impl()
-  if not soabi and impl in ('cp', 'pp') and hasattr(sys, 'maxunicode'):
-    d = ''
-    m = ''
-    u = ''
-    if get_flag('Py_DEBUG',
-                lambda: hasattr(sys, 'gettotalrefcount'),
-                warn=(impl == 'cp')):
-      d = 'd'
-    if get_flag('WITH_PYMALLOC',
-                lambda: impl == 'cp',
-                warn=(impl == 'cp')):
-      m = 'm'
-    if (get_flag('Py_UNICODE_SIZE',
-                 lambda: sys.maxunicode == 0x10ffff,
-                 expected=4,
-                 warn=(impl == 'cp' and sys.version_info < (3, 3))) and
-        sys.version_info < (3, 3)):
-      u = 'u'
-    abi = '%s%s%s%s%s' % (impl, get_impl_ver(), d, m, u)
-  elif soabi and soabi.startswith('cpython-'):
-    abi = 'cp' + soabi.split('-')[1]
-  elif soabi:
-    abi = soabi.replace('.', '_').replace('-', '_')
-  else:
-    abi = None
-  return abi
+    """Return the ABI tag based on SOABI (if available) or emulate SOABI
+    (CPython 2, PyPy)."""
+    soabi = get_config_var('SOABI')
+    impl = get_abbr_impl()
+    if not soabi and impl in {'cp', 'pp'} and hasattr(sys, 'maxunicode'):
+        d = ''
+        m = ''
+        u = ''
+        if get_flag('Py_DEBUG',
+                    lambda: hasattr(sys, 'gettotalrefcount'),
+                    warn=(impl == 'cp')):
+            d = 'd'
+        if get_flag('WITH_PYMALLOC',
+                    lambda: impl == 'cp',
+                    warn=(impl == 'cp')):
+            m = 'm'
+
+        # NB: Modified from ~ `from .extern import six; six.PY2`
+        PY2 = sys.version_info[0] == 2
+        if (get_flag('Py_UNICODE_SIZE',
+                     lambda: sys.maxunicode == 0x10ffff,
+                     expected=4,
+                     warn=(impl == 'cp' and PY2)) and PY2):
```

Comment:
To confirm, we ignore a check for 3.0-3.2 because Pex doesn't support these versions, right? Consider adding a brief comment.

```
NB: this block is modified to only check if PY2, rather than <3.3, because PEX does not support 3.0-3.2.
```

Reply:
Nack. The file on the lhs is older than on the rhs. My modification was only to inline setuptools's vendored six.PY2 use here.